play-ws-standalone/src/main/java/play/libs/ws/StandaloneWSResponse.java

```diff
@@ -122,6 +122,15 @@
      *
      * Note that this is only usable with a streaming request:
      *
+     * <pre>
+     * {@code
+     * wsClient.url("https://playframework.com")
+     *         .stream() // this returns a CompletionStage<StandaloneWSResponse>
+     *         .thenApply(StandaloneWSResponse::getBodyAsSource);
+     * }
+     * </pre>
      */
-    Source<ByteString, ?> getBodyAsSource();
+    default Source<ByteString, ?> getBodyAsSource() {
+        return Source.single(getBodyAsBytes());
```

Comment:
There's `WSResponse` and `StreamedResponse`. One is optimized for strict bodies and the other for streamed bodies. It's definitely not the greatest API, but it's consistent with the way WS has worked in the past.